File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/presentation/paths/statistics/StatisticsFragment.java
package com.zhuinden.simplestackdemoexamplefragments.presentation.paths.statistics;

import android.content.res.Resources;
import android.view.View;
import android.widget.TextView;

import com.zhuinden.simplestackdemoexamplefragments.R;
import com.zhuinden.simplestackdemoexamplefragments.application.Injector;
import com.zhuinden.simplestackdemoexamplefragments.util.BaseFragment;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by Owner on 2017. 01. 26..
 */
// UNSCOPED!
public class StatisticsFragment
        extends BaseFragment<StatisticsFragment, StatisticsPresenter> {
    public StatisticsFragment() {
    }

    private Resources resources;
    private StatisticsPresenter statisticsPresenter;

    @BindView(R.id.statistics)
    TextView mStatisticsTV;

    @Override
    public StatisticsPresenter getPresenter() {
        return statisticsPresenter;
    }

    @Override
    public StatisticsFragment getThis() {
        return this;
    }

    @Override
    protected Unbinder bindViews(View view) {
        return ButterKnife.bind(this, view);
    }

    @Override
    protected void injectSelf() {
        resources = Injector.get()
                .resources();
        statisticsPresenter = Injector.get()
                .statisticsPresenter();
    }


    public void setProgressIndicator(boolean active) {
        if(active) {
            mStatisticsTV.setText(resources.getString(R.string.loading));
        } else {
            mStatisticsTV.setText("");
        }
    }

    public void showStatistics(int numberOfIncompleteTasks, int numberOfCompletedTasks) {
        if(numberOfCompletedTasks == 0 && numberOfIncompleteTasks == 0) {
            mStatisticsTV.setText(resources.getString(R.string.statistics_no_tasks));
        } else {
            String displayString = resources.getString(R.string.statistics_active_tasks) + " " + numberOfIncompleteTasks + "\n" + resources.getString(
                    R.string.statistics_completed_tasks) + " " + numberOfCompletedTasks;
            mStatisticsTV.setText(displayString);
        }
    }

    public void showLoadingStatisticsError() {
        mStatisticsTV.setText(resources.getString(R.string.statistics_error));
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/presentation/paths/statistics/StatisticsFragment.kt
package com.zhuinden.simplestackdemoexamplefragments.presentation.paths.statistics

import com.zhuinden.simplestackdemoexamplefragments.R
import com.zhuinden.simplestackdemoexamplefragments.application.Injector
import com.zhuinden.simplestackdemoexamplefragments.util.BaseFragment
import com.zhuinden.simplestackdemoexamplefragments.util.MvpPresenter
import com.zhuinden.simplestackdemoexamplefragments.util.backstackDelegate
import kotlinx.android.synthetic.main.path_statistics.*

/**
 * Created by Zhuinden on 2018. 08. 20.
 */
// UNSCOPED!
class StatisticsFragment : BaseFragment<StatisticsFragment, StatisticsFragment.Presenter>() {
    private val myResources = Injector.get().resources()

    companion object {
        const val CONTROLLER_TAG = "AddOrEditTaskPresenter"
    }

    interface Presenter: MvpPresenter<StatisticsFragment> {
    }

    override val presenter: Presenter by lazy {
        backstackDelegate.lookupService<StatisticsFragment.Presenter>(CONTROLLER_TAG)
    }

    override fun getThis(): StatisticsFragment = this

    fun setProgressIndicator(active: Boolean) {
        textStatistics.text = when {
            active -> myResources.getString(R.string.loading)
            else -> ""
        }
    }

    fun showStatistics(numberOfIncompleteTasks: Int, numberOfCompletedTasks: Int) {
        textStatistics.text = when {
            numberOfCompletedTasks == 0 && numberOfIncompleteTasks == 0 -> myResources.getString(R.string.statistics_no_tasks)
            else ->
                "${myResources.getString(R.string.statistics_active_tasks)} $numberOfIncompleteTasks\n" +
                    "${myResources.getString(R.string.statistics_completed_tasks)} $numberOfCompletedTasks"
        }
    }

    fun showLoadingStatisticsError() {
        textStatistics.text = myResources.getString(R.string.statistics_error)
    }
}
